src/main/java/crawlercommons/sitemaps/SiteMapRSSSAXHandler.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package crawlercommons.sitemaps;
+
+import static crawlercommons.sitemaps.SiteMapParser.LOG;
+import static crawlercommons.sitemaps.SiteMapParser.urlIsValid;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.LinkedList;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+import crawlercommons.sitemaps.AbstractSiteMap.SitemapType;
+
+/**
+ * Parse XML document which is assumed to be in RSS format. RSS 2.0 example:
+ * 
+ * <?xml version="1.0"?> <rss version="2.0"> <channel> <title>Lift Off
+ * News</title> <link>http://liftoff.msfc.nasa.gov/</link> <description>Liftoff
+ * to Space Exploration.</description> <language>en-us</language> <pubDate>Tue,
+ * 10 Jun 2003 04:00:00 GMT</pubDate> <lastBuildDate>Tue, 10 Jun 2003 09:41:01
+ * GMT</lastBuildDate> <docs>http://blogs.law.harvard.edu/tech/rss</docs>
+ * <generator>Weblog Editor 2.0</generator>
+ * <managingEditor>editor@example.com</managingEditor>
+ * <webMaster>webmaster@example.com</webMaster> <ttl>5</ttl>
+ * 
+ * <item> <title>Star City</title>
+ * <link>http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp</link>
+ * <description>How do Americans get ready to work with Russians aboard the
+ * International Space Station? They take a crash course in culture, language
+ * and protocol at Russia's Star City.</description> <pubDate>Tue, 03 Jun 2003
+ * 09:39:21 GMT</pubDate>
+ * <guid>http://liftoff.msfc.nasa.gov/2003/06/03.html#item573</guid> </item>
+ * 
+ * <item> <title>Space Exploration</title>
+ * <link>http://liftoff.msfc.nasa.gov/</link> <description>Sky watchers in
+ * Europe, Asia, and parts of Alaska and Canada will experience a partial
+ * eclipse of the Sun on Saturday, May 31.</description> <pubDate>Fri, 30 May
+ * 2003 11:06:42 GMT</pubDate>
+ * <guid>http://liftoff.msfc.nasa.gov/2003/05/30.html#item572</guid> </item>
+ * 
+ * </channel> </rss>
+ */
+class SiteMapRSSSAXHandler extends AbstractSiteMapSAXHandler {
+
+    private SiteMap sitemap;
+    private URL loc;
+    private String lastMod;
+    boolean valid;
+    private int i = 0;
+
+    SiteMapRSSSAXHandler(URL url, LinkedList<String> elementStack, boolean strict) {
+        super(elementStack, strict);
+        sitemap = new SiteMap(url);
+        sitemap.setType(SitemapType.RSS);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * crawlercommons.sitemaps.AbstractSiteMapSAXHandler#startElement(java.lang
+     * .String, java.lang.String, java.lang.String, org.xml.sax.Attributes)
+     */
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
```

Comment:
Doesn't this need to call super.startElement? Otherwise the element stack isn't going to be set up, right?